### DISCUSSION TITLE
How to play the audio in web browser and in jamulus at the same time?

### DISCUSSION THREAD

**Discussion by ThulithaNawagamuwa** [jamulussoftware/jamulus#2539] [Technical Support & Help]
As we tried, we couldn't hear the sound from other applications on the computer while running the Jamulus.

For our project, we **need to play the sound in the web browser and the Jamulus at the same time**. We are running a virtual metronome in the web browser.

Does anyone know how to do this?

Huge thanks.

**Comment by jujudusud** [comment]
Hi,
Is it on macos? Windows? Linux?

**Reply by ThulithaNawagamuwa** [reply]
On windows

**Comment by ann0see** [comment]
Hi @ThulithaNawagamuwa yes, it’s totally possible to get sound into Jamulus. The people on WorldJam do it all the time. From my experience Linux is easiest, afterwards macOS and then Windows. For Windows maybe my article helps: https://jamulus.io/kb/2022/01/26/Routing-JACK-Windows.html

feel free to improve it if necessary. 

Some people say that in fact iOS is easier — but I don‘t have experience with that unfortunately.

**Comment by ann0see** [comment]
If you don't want to put the audio of the Browser directly into Jamulus it's easier - but probably impossible with ASIO4ALL, if you're using that. You can use an audio interface which supports ASIO and normal Windows audio at the same time (I think even cheep ones support that like the UMC202HD from Behringer). Otherwise, consider using another output or device for your browser.

**Reply by ThulithaNawagamuwa** [reply]
Thanks for the idea. Currently, I'm using Focusrite Scarlett Solo audio interface for the project. I'm not sure whether it supports both types of audio at the same time.

**Comment by pgScorpio** [comment]
On Windows the 'easiest' solution would be using ASIO Link Pro. But because ALP is somewhat hard to master I have created an "all-in-one" installer (originally intended to link Jamulus and Zoom audio) that also provides in all necessary shortcuts and ALP configuration files to accomplish this.

See my [JamZoom-ALP installer](http://peegee.dsmynas.net:8012/fbdownload/JamZoom-ALP?k=4jlf4yZ4&stdhtml=true) which can also be found in the files section of the Jamulus Choral Community on Facebook.
